solutions/cf-1627E.mdx

```diff
@@ -0,0 +1,147 @@
+---
+id: cf-1627E
+source: CF
+title: Not Escaping
+author: Chongtian Ma
+---
+
+[Official Analysis (C++, Java, and Python)](https://codeforces.com/blog/entry/99067)
+
+## Alternate Explanation
+
+There can be at most $n \cdot m$ rooms, which is about $10^{10}$ total rooms and too much to process. However, not all of them are relevant (i.e. there are some rooms that are not a start point or an end point for a ladder, so processing them would be extraneous). Therefore, coordinate compression can be used here.
+
+We can use shortest paths to solve the problem. Since we can move freely on a floor, we can perform dijkstra to find the minimal health to reach every room on that floor. Then, we can iterate over all the ladders on that floor and calculate the minimum health to travel to the endpoints of each ladder by using it.
```

Comment:
```suggestion
We can use shortest paths to solve the problem. Since we can move freely on a floor, we can use Dijkstra's to find the minimal health to reach every room on that floor. Then, we can iterate over all the ladders on that floor and calculate the minimum health to travel to the endpoints of each ladder by using it.
```